asciigraf/asciigraf.py

```diff
@@ -41,52 +47,107 @@ def graph_from_ascii(network_string):
-        # mapped the character to an edge
-        if pos not in edge_char_to_edge_map:
-            if neighbor in edge_char_to_edge_map:  # Add this node to the edge
-                edge_char_to_edge_map[pos] = edge_char_to_edge_map[neighbor]
-                edge_char_to_edge_map[pos]["points"].append(pos)
-            else:  # Make a new edge
-                edge_char_to_edge_map[pos] = dict(points=[pos], nodes=[])
-                edges.append(edge_char_to_edge_map[pos])
-
-            # we can look ahead at other neighbor and add it too -- this
-            # step allows us to solve a few extra corner types
-            if (
-                neighbor_2 not in edge_char_to_edge_map
-                and neighbor_2 in edge_chars
-            ):
-                edge_char_to_edge_map[neighbor_2] = edge_char_to_edge_map[pos]
-                edge_char_to_edge_map[pos]["points"].append(neighbor_2)
-
-        neighboring_nodes = [
-            node_char_to_node[pos+pos_offset]
-            for pos_offset in EDGE_CHAR_NEIGHBOURS[char]
-            if pos+pos_offset in node_char_to_node
-        ]
-        edge_char_to_edge_map[pos]["nodes"] += neighboring_nodes
-
-    # Make the networkx graph
-    for edge in edges:
-        edge_map = {pos: edge_chars[pos] for pos in edge["points"]}
-        if len(edge['nodes']) > 2:
-            raise InvalidEdgeError(
-                "Too many nodes:\nNetwork String:\n" +
-                "{}\nAffected Edge:\n{}".format(network_string, draw(edge_map))
-            )
-        elif len(edge['nodes']) < 2:
+    def get_abutting_edge_chars(pos):
+        """ Return the edge/node positions that neighbour the given position.
+
+            e.g.
+             ___
+            |  /|
+            | *-|    -> Point(3, 1), Point(3, 2) are neighbours
+            |___|
+             ___
+            |  /|
+            |-* |    -> Point(3, 1), Point(1, 2) are neighbours
+            |___|
+             ___
+            |  -|
+            |---|    -> Point(1, 2), Point(3, 2) are neighbours
+            |___|
+             ___
+            |  -|
+            |--Node| -> Point(1, 2), Point(3, 2) are neighbours
+            |___|
+        """
+        neighbouring_positions = set()
+
+        # first, consider neighbours of our char (e.g. if our char
+        # is '-' then any node or edge char to the left or right
+        # is neighbouring to the char at `pos`)
+        for offset in EDGE_CHAR_NEIGHBOURS[edge_chars[pos]]:
+            neighbour = pos + offset
+            if neighbour in edge_chars or neighbour in node_char_to_node:
+                neighbouring_positions |= {neighbour}
+
+        # second, consider chars to which this char could be a neighbour
+        # (e.g. if the char below is a |, our char neighbours it)
+        for offset, valid_char in ABUTTING.items():
+            if edge_chars.get(pos + offset, " ") == valid_char:
+                neighbouring_positions |= {pos + offset}
```

Comment:
I'm having a hard time wrapping my mind around the difference between `EDGE_CHAR_NEIGHBOURS` and `ABUTTING`